src/proofs/Term.cpp

```diff
@@ -192,12 +292,12 @@ std::shared_ptr<Term> RemoveUnusedVisitor::visit(Quant * term) {
 
     for (std::size_t i = 0; i < vars.size(); i++) {
         auto var = vars[i];
-        for (auto varInUse : varsInUse) {
+        for (const auto & varInUse : varsInUse) {
             if (var->printTerm() == varInUse) { inUse = true; }
         }
         if (!inUse) {
-            vars.erase(vars.begin() + i);
-            sorts.erase(sorts.begin() + i);
+            vars.erase(vars.begin() + int(i));
+            sorts.erase(sorts.begin() + int(i));
```

Comment:
I don't see the need to cast this to `int`. What am I missing?
```suggestion
            vars.erase(vars.begin() + i);
            sorts.erase(sorts.begin() + i);
```

Reply:
It gives a warning about narrowing conversion.

Reply:
OK, I see. While looking at this code, I noticed that it actually does not work correctly.
I'll write a suggestion in a separate comment.